### ISSUE TITLE
Need a function which checks whether a date is always just a date and does not contain time information

### ISSUE THREAD

**Issue by RobinL** [moj-analytical-services/data_linter#11]
i.e. distinguishes between a column in the metadata being specified as  `datetime` and `date`

**Comment by isichei** [comment]
Think this needs something like `check_date_precision` or something?

- One check will be to make sure that say that if your (date) data is coming in as `2019-01-01 00:00:00` then the time is always 00:00:00 for dates?
- What do we do with columns defined as dates but do have time information. Even if all your dates end with `00:00:00` you would probably want to customise a response that may error/warn the users that time information does exist even though it is a date specification (even if everything in the sample has 0 time it may not do in the future). 
- I'm not sure if we need to check the data in strong format for (csv and jsonl). Do you think it is necessary to see what the precision is of the data in string format? Maybe something like:

```
# For jsonl/csv
-> Read in dates as strings
-> Find minimum precision (e.g. yyyy-mm-dd or yyyy-mm-dd HH:MM:SS)
-> Then test if all vals conform to minimum precision or are they mixed?
```

The above should be possible - this comes back to the issue we discussed before though - if data_linter should manage reading in data.

**Comment by RobinL** [comment]
Yes - not sure what to do in the case you see time information but coltype is date.  

ATM the type check is a check of whether we can impose a type on the column - e.g. can we impose type `int` on value `"hello"`.  Question is really whether that should succeed.  

python's `datetime` does allow a time to be converted into a date without complaint: `import datetime; n = datetime.datetime.now(); n.date()`.  

you can also do `int(1.21)` without error 

We also allow it:
```
import pandas as pd
from data_linter.lint import Linter

meta = {'columns': [{'name': 'floats', 'type': 'int'}]}
df = pd.DataFrame({"floats": [1.1, 1.2]})
l = Linter(df, meta)
l.check_all()
``` 
check_all is successful

**Comment by RobinL** [comment]
My view is that there should be an optional strict mode which makes the linter fail when you lose precision (float -> int, datetime -> date, long -> int)
